sds/src/proximity_keywords.rs

```diff
@@ -317,6 +346,101 @@ fn calculate_keyword_pattern(keyword: &str) -> Ast {
     })
 }
 
+#[derive(Clone, Copy)]
+enum CharType {
+    Regular,
+    Uppercase,
+    Separator,
+}
+
+fn get_chars_type(prev: &char, current: &char, next: &char) -> (CharType, CharType, CharType) {
+    let get_single_char_type = |c: &char| -> CharType {
+        let is_link_symbol = MULTI_WORD_KEYWORDS_LINK_CHARS.contains(c);
+        let is_uppercase_char = c.is_ascii_uppercase();
+
+        if is_link_symbol {
+            CharType::Separator
+        } else if is_uppercase_char {
+            CharType::Uppercase
+        } else {
+            CharType::Regular
+        }
+    };
+
+    (
+        get_single_char_type(prev),
+        get_single_char_type(current),
+        get_single_char_type(next),
+    )
+}
+
+fn standardize_path_chars<F>(chars: Vec<char>, mut push_character: F)
+where
+    F: FnMut(&char),
+{
+    let kw_length = chars.len();
+
+    if kw_length >= 1 {
+        push_character(&chars[0])
+    }
+
+    for chars in chars.windows(3) {
```

Comment:
While reading the code, it was not clear to me why this was needed at first. I think it would be easier to read if the condition was the contrary:
```suggestion
    if chars.is_empty(){
        return;
    }

    push_character(&chars[0]);
    for chars in chars.windows(3) {
```